### DISCUSSION TITLE
Programmatically add custom sub-domain of a custom domain

### DISCUSSION THREAD

**Discussion by ghubnab99** [vercel/community#3231] [Help] [answered]
### Question

Hi,

We have a custom domain (say for eg. techno.com) that is registered with Godaddy.

We are developing a multi tenant application where during the user registration, he should be able to specify a store name (say for eg. styleshoes) and a subdomain should be programmatically be created like (styleshoes.techno.com).

And he should be able to login to his admin section at: styleshoes.techno.com/admin

How can we handle this?

Thanks

**Comment by swarnava** [comment, marked as answer]
Hi,

We have a guide article that may help you  to get started: https://vercel.com/guides/nextjs-multi-tenant-application

**Reply by ghubnab99** [reply]
Yes, I've already gone through that but couldn't find anything specific to our requirement

**Reply by amyegan** [reply]
I think Swarnava is right. You should take another look at the Platforms Starter Kit resources. 

Specifically, I think the section describing how to [deploy a multi-tenant app to Vercel](https://vercel.com/guides/nextjs-multi-tenant-application#5.-deploy-to-vercel) is what you're looking for. The [wildcard domain](https://vercel.com/blog/wildcard-domains) setup will let you use subdomains without setting them up individually.

You can find an example of how to route traffic based on subdomain in the [Platforms Starter Kit](https://github.com/vercel/platforms) repo. [Middleware](https://nextjs.org/docs/app/building-your-application/routing/middleware) and [dynamic routes](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes) are they key things to help with that.
